src/ring_theory/dedekind_domain.lean

```diff
@@ -592,4 +593,10 @@ noncomputable instance fractional_ideal.comm_group_with_zero
   mul_inv_cancel := λ I, fractional_ideal.mul_inv_cancel,
   .. fractional_ideal.comm_semiring }
 
+noncomputable instance ideal.comm_cancel_monoid_with_zero
+  [is_dedekind_domain A] : comm_cancel_monoid_with_zero (ideal A) :=
+{ .. function.injective.cancel_monoid_with_zero (coe_ideal_hom A⁰ (fraction_ring A))
+       coe_ideal_injective (ring_hom.map_zero _) (ring_hom.map_one _) (ring_hom.map_mul _),
+  .. (infer_instance : comm_monoid_with_zero (ideal A)) }
```

Comment:
Can you add the missing `function.injective.comm_monoid_with_zero` definition that you're working around here? It should be as trivial as the implementation you have here.